static/app/views/issueDetails/streamline/useFetchEvents.tsx

```diff
@@ -0,0 +1,68 @@
+import {getInterval} from 'sentry/components/charts/utils';
+import type {Group} from 'sentry/types/group';
+import type {MultiSeriesEventsStats} from 'sentry/types/organization';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {getPeriod} from 'sentry/utils/duration/getPeriod';
+import {getConfigForIssueType} from 'sentry/utils/issueTypeConfig';
+import {
+  type ApiQueryKey,
+  useApiQuery,
+  type UseApiQueryOptions,
+} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {useEnvironmentsFromUrl} from 'sentry/views/issueDetails/utils';
+
+interface UseFetchEventStatsParameters {
+  group: Group;
+  referrer: string;
+  query?: string;
+}
+
+function makeFetchEventStatsQueryKey({
+  organizationSlug,
+  query,
+}: {
+  organizationSlug: string;
+  query: Record<string, any>;
+}): ApiQueryKey {
+  return [`/organizations/${organizationSlug}/events-stats/`, {query}];
+}
+
+export function useFetchEventStats({
+  params: {group, query, referrer},
+  options,
+}: {
+  params: UseFetchEventStatsParameters;
+  options?: UseApiQueryOptions<MultiSeriesEventsStats>;
+}) {
+  const organization = useOrganization();
+  const {selection: pageFilters} = usePageFilters();
+  const environment = useEnvironmentsFromUrl() ?? pageFilters.environments;
+  const periodQuery = getPeriod(pageFilters.datetime);
+  const interval = getInterval(pageFilters.datetime, 'low');
+  const config = getConfigForIssueType(group, group.project);
+  const fullQuery = {
+    ...periodQuery,
+    interval,
+    referrer,
+    environment,
+    yAxis: ['count()', 'count_unique(user)'],
+    dataset: config.usesIssuePlatform
+      ? DiscoverDatasets.ISSUE_PLATFORM
+      : DiscoverDatasets.ERRORS,
+    project: group.project.id,
+    query: query,
+  };
+  const queryKey = makeFetchEventStatsQueryKey({
+    organizationSlug: organization.slug,
+    query: fullQuery,
+  });
+
+  return useApiQuery<MultiSeriesEventsStats>(queryKey, {
+    staleTime: 30000,
+    cacheTime: 30000,
```

Comment:
just leave cache time to the default of 5 minutes if its stale it'll refetch in the background anyway
```suggestion
    staleTime: 30000,
```